app/frontend/components/pages/txHistory/transactionHistory.tsx

```diff
@@ -188,35 +214,102 @@ const ExportCSV = ({transactionHistory, stakingHistory}) => {
   )
 }
 
-class TransactionHistory extends Component<Props> {
-  render({transactionHistory, stakingHistory}) {
-    const formatDate = (date) => toLocalDate(new Date(date * 1000))
-
-    return (
-      <div className="transactions card">
-        <div className="transactions-header">
-          <h2 className="card-title">Transaction History</h2>
-          <div className="download-transaction">
-            <ExportCSV transactionHistory={transactionHistory} stakingHistory={stakingHistory} />
-          </div>
+const throtle = (fn, ms) => {
+  let timer
+  return () => {
+    if (timer) return
+    timer = setTimeout(() => {
+      timer = null
+      fn()
+    }, ms)
+  }
+}
+
+// TODO: do something with magic numbers?
+const calculateMaEllipsisCount = (): number | null => {
```

Comment:
I'm not sure how to do ellipsis in the middle of word (hash) in pure css. This function redefines some values that are derived from styles, like `TABLET_BREAKPOINT_PX`. Any ideas to make this more elegant are appreciated.